pages/common/xargs.md

```diff
@@ -1,19 +1,20 @@
 # xargs
 
-> Execute a command with piped arguments.
+> Execute a command with piped arguments coming from another command, a file, etc.
+> The input is treated as a single block of text and split into separate arguments on spaces, tabs, newlines and end-of-file.
 
-- Main use:
+- Main usage pattern:
 
-`{{arguments}} | xargs {{command}}`
+`{{arguments_source}} | xargs {{command}}`
 
-- Specific example: delete all files that start with 'M':
+- Delete all files with a `.backup` extension:
 
-`find . -name 'M*' | xargs rm`
+`{{find . -name '*.backup'}} | xargs {{rm -v}}`
 
-- Handle whitespace in arguments:
+- Use the NUL (`\0`) character as separator (useful to split the input on newlines but not on spaces):
```

Comment:
Re-reading @ostera's comment with a fresh mind, I understand the value on specifying that the input is split on newlines, which isn't indicated on neither mine nor @agnivade's proposals. I would like to come up with a good compromise. Any suggestions?